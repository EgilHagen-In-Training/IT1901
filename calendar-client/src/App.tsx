import React, { FC } from "react";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import { ChakraProvider, extendTheme } from "@chakra-ui/react";
import Login from "./pages/Login";
import Calendar from "./pages/Calendar";
import Event from "./pages/Event";

const theme = extendTheme({
	styles: {
		global: {
			body: {
				bg: "yellow.100",
				color: "black",
			},
		},
	},
	initialColorMode: "dark",
	useSystemColorMode: false,
});

const App: FC = () => {
	return (
		<ChakraProvider theme={theme}>
			<Router>
				<Switch>
					<Route path="/event">
						<Event />
					</Route>
					<Route path="/calendar">
						<Calendar />
					</Route>
					<Route path="/">
						<Login />
					</Route>
				</Switch>
			</Router>
		</ChakraProvider>
	);
};

export default App;
