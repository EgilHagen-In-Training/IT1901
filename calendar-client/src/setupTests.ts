// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom";
import { User } from "./types/user";

export const mockUser: User = {
	username: "testUser",
	calendar: {
		events: [
			{
				header: "testEvent1",
				description: "testDescription1",
				date: "2021-11-23",
				timeString: "12:30",
				timeHour: 12,
				timeMinute: 30,
				id: 1,
			},
			{
				header: "testEvent2",
				description: "testDescription1",
				date: "2021-11-11",
				timeString: "15:00",
				timeHour: 15,
				timeMinute: 0,
				id: 2,
			},
		],
	},
};
