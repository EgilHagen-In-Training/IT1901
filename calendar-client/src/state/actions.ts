import { User } from "../types/user";

export enum ActionTypes {
  SET_USER = "SET_USER",
  LOGOUT_USER = "LOGOUT_USER",
}

type SetUserAction = { type: ActionTypes.SET_USER; payload: User };
type LogoutUserAction = { type: ActionTypes.LOGOUT_USER };


export type Action =
  | SetUserAction
  | LogoutUserAction

export const setUser = (user: User): SetUserAction => ({
	type: ActionTypes.SET_USER,
	payload: user,
});

export const logoutUser = (): LogoutUserAction => ({
	type: ActionTypes.LOGOUT_USER,
});

