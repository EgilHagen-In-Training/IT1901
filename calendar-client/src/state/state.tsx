import React, { createContext, Dispatch, ReactNode, useReducer } from "react";
import { User, EventType } from "../types/user";
import { Action, ActionTypes } from "./actions";

type State = {
	user?: User;
	currentEvent?: EventType;
};

const initialState: State = {
	user: undefined,
	currentEvent: undefined,
};

const StateReducer = (state: State, action: Action): State => {
	switch (action.type) {
		case ActionTypes.SET_USER:
			return { ...state, user: action.payload };
		case ActionTypes.LOGOUT_USER:
			return {
				...state,
				user: undefined,
			};
		case ActionTypes.SET_EVENT:
			return { ...state, currentEvent: action.payload };
		case ActionTypes.CLEAR_EVENT:
			return { ...state, currentEvent: undefined };
		default:
			return { ...state };
	}
};

type StateContextProps = {
	state: State;
	dispatch: Dispatch<Action>;
};

const initialContext: StateContextProps = {
	state: initialState,
	dispatch: () => null,
};

export const StateContext = createContext<StateContextProps>(initialContext);

type StateProviderProps = {
	children: ReactNode;
};

export const StateProvider = (props: StateProviderProps): JSX.Element => {
	const [state, dispatch] = useReducer(StateReducer, initialState);
	const { children } = props;

	return (
		<StateContext.Provider value={{ state, dispatch }}>
			{children}
		</StateContext.Provider>
	);
};
