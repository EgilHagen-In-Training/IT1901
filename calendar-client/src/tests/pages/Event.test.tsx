import React from "react";
import App from "../../App";
import renderer from "react-test-renderer";
import { StateContext } from "../../state/state";
import { mockUser } from "../../setupTests";
import { Router } from "react-router";
import { createMemoryHistory } from "history";

test("Event renders correctly", () => {
	const state = {
		user: mockUser,
	};
	const dispatch = jest.fn();
	const history = createMemoryHistory();
	history.push("/event");

	const tree = renderer
		.create(
			<StateContext.Provider value={{ state, dispatch }}>
				<Router history={history}>
					<App />
				</Router>
			</StateContext.Provider>
		)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
