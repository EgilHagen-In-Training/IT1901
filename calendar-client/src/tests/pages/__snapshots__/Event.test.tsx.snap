// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event renders correctly 1`] = `
Array [
  <div
    className="css-8atqhb"
  >
    <button
      aria-label="logout button"
      className="chakra-button css-7yzoig"
      onClick={[Function]}
      type="button"
    >
      Log out
    </button>
    <h2
      className="chakra-heading css-xrxou1"
    >
      november
       
      2021
    </h2>
    <div
      className="css-gmuwbf"
    >
      <div
        className="chakra-stack css-mxsw1"
      >
        <button
          className="chakra-button css-taj3dd"
          onClick={[Function]}
          type="button"
        >
          <span
            className="chakra-button__icon css-1wh2kri"
          >
            <svg
              aria-hidden={true}
              className="chakra-icon css-onkibi"
              focusable={false}
              viewBox="0 0 24 24"
            >
              <path
                d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
                fill="currentColor"
              />
            </svg>
          </span>
          Forige måned
        </button>
        <button
          className="chakra-button css-taj3dd"
          onClick={[Function]}
          type="button"
        >
          I dag
        </button>
        <button
          className="chakra-button css-taj3dd"
          onClick={[Function]}
          type="button"
        >
          Neste måned
          <span
            className="chakra-button__icon css-1hzyiq5"
          >
            <svg
              aria-hidden={true}
              className="chakra-icon css-onkibi"
              focusable={false}
              viewBox="0 0 24 24"
            >
              <path
                d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
                fill="currentColor"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
    <hr
      aria-orientation="horizontal"
      className="chakra-divider css-impwoj"
    />
    <div
      className="css-nvbfv9"
    >
      <div
        className="css-0"
      >
        Mandag
      </div>
      <div
        className="css-0"
      >
        Tirsdag
      </div>
      <div
        className="css-0"
      >
        Onsdag
      </div>
      <div
        className="css-0"
      >
        Torsdag
      </div>
      <div
        className="css-0"
      >
        Fredag
      </div>
      <div
        className="css-0"
      >
        Lørdag
      </div>
      <div
        className="css-0"
      >
        Søndag
      </div>
    </div>
    <div
      className="css-gmuwbf"
    >
      <div
        className="css-1vaoh93"
      />
    </div>
    <hr
      aria-orientation="horizontal"
      className="chakra-divider css-impwoj"
    />
    <div
      className="css-gmuwbf"
    >
      <button
        className="chakra-button css-taj3dd"
        onClick={[Function]}
        type="button"
      >
        Ny hendelse
      </button>
    </div>
  </div>,
  <span />,
]
`;
